lib/resty/rediscluster.lua

```diff
@@ -412,8 +412,7 @@ end
 local function handle_command_with_retry(self, target_ip, target_port, asking, cmd, key, ...)
     local config = self.config
 
-    key = tostring(key)
-    local slot = redis_slot(key)
```

Comment:
The `key` could be an arg passed to the `lua-resty-redis` client, and it could be a Lua table. 
So only cast to string when calling `redis_slot` is right.

https://github.com/Kong/resty-redis-cluster/blob/30a11df4cc253de05845e63bd9eda9b6ba608c6c/lib/resty/rediscluster.lua#L669